code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -274,6 +283,8 @@
 		if(ranged) //We ranged? Shoot at em
 			if(!target.Adjacent(targets_from) && ranged_cooldown <= world.time) //But make sure they're not in range for a melee attack and our range attack is off cooldown
 				OpenFire(target)
+		//Component handles all the math and charge handling, so we can just send the signal here.
+		SEND_SIGNAL(src, COMSIG_HOSTILE_CHARGING_TARGET, target)
```

Comment:
Diagnosing this now but I don't have the slighest clue why this is, so open to suggestions:
Charges are currently breaking because target is... itself. This target seems to work for the purposes of regular pathfinding, but when brought into the component it's causing the mob to throw itself at... itself.

Reply:
I believe SEND_SIGNAL tacks the target (in this case src) to the beginning of the passed args. The datum relaying the signal will always be the first argument passed to any listeners. In this case specifically the args being passed are (src, target).

Reply:
https://github.com/tgstation/tgstation/blob/b7d960b18727627287eb7ec2d8c0d712a4e8c3a7/code/__DEFINES/dcs/helpers.dm#L1-L5